### PR TITLE
Kiali does not handle namespace-level injection enable/disable for Maistra

### PR DESCRIPTION
**Pull request by jshaughn**
Update Cypress to handle that use-case.

https://issues.redhat.com/browse/OSSM-3989

### CODE REVIEW

File: frontend/cypress/integration/common/sidecar_injection.ts [hunk 1]
```diff
@@ -239,32 +239,64 @@ When('I visit the overview page', function () {
...
+    expect(response.status).to.equal(200);
+    const isMaistra = response.body.istioEnvironment.isMaistra;
+
+    cy.get('[data-test=overview-type-LIST]').should('be.visible').click();
+    cy.get('[data-test=overview-type-LIST]').should('be.visible').click();
+    cy.get(`[data-test=VirtualItem_${this.targetNamespace}] button`).should('be.visible').click();
+    // Maistra does not support namespace-level injection
+    if (isMaistra) {
+      cy.get(`[data-test=enable-${this.targetNamespace}-namespace-sidecar-injection]`).should('not.exist');
+    } else {
```
Comment (nrfox) [inline frontend/cypress/integration/common/sidecar_injection.ts:252]: These actually seem like two separate tests. You don't need to include this with this change but it'd be nice to start using tags to include/exclude certain tests based on the env. We'll need to do this for multi-cluster tests and probably other things as well.

**Review by leandroberetta — APPROVED** [review]
Looks good.

File: frontend/cypress/integration/common/sidecar_injection.ts [hunk 2]
```diff
@@ -239,32 +239,64 @@ When('I visit the overview page', function () {
...
 
 When('I override the default automatic sidecar injection policy in the namespace to enabled', function () {
-  cy.get('[data-test=overview-type-LIST]').should('be.visible').click();
-  cy.get(`[data-test=VirtualItem_${this.targetNamespace}] button`).should('be.visible').click();
-  cy.get(`[data-test=enable-${this.targetNamespace}-namespace-sidecar-injection]`).should('be.visible').click();
-  cy.get('[data-test=confirm-traffic-policies]').should('be.visible').click();
-  ensureKialiFinishedLoading();
+  cy.request('GET', '/api/status').should(response => {
+    expect(response.status).to.equal(200);
+    const isMaistra = response.body.istioEnvironment.isMaistra;
```
Comment (nrfox) [inline frontend/cypress/integration/common/sidecar_injection.ts:244]: It'd be nice to get this kind of info once before any of the tests are run but I'm not sure the best way of doing that within cypress.

**Review by nrfox — APPROVED** [review]
Just some comments for the future but otherwise LGTM

File: frontend/cypress/integration/common/sidecar_injection.ts (hunk 1 above)
Comment (jshaughn) [reply frontend/cypress/integration/common/sidecar_injection.ts:252]: Yes, at the moment just trying to get the stuff passing so it's not blocking 2.4.  With 3.0 it's possible things will align with upstream and we won't have the disparity.

File: frontend/cypress/integration/common/sidecar_injection.ts (hunk 2 above)
Comment (jshaughn) [reply frontend/cypress/integration/common/sidecar_injection.ts:244]: me either...

**Pull request merged**